### PR TITLE
MGMT-808: add reason to all HTTP errors

### PR DESCRIPTION
**Pull request by filanov**
temporary adding the full error without filtering, later will filter out only the relevant error message for the user

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -500,34 +499,34 @@ func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer
...
 		tx.Rollback()
 		log.WithError(err).Errorf("failed to validate or update cluster %s state", params.ClusterID)
 		return installer.NewRegisterClusterInternalServerError().
-			WithPayload(generateError(http.StatusInternalServerError))
+			WithPayload(generateError(http.StatusInternalServerError, err))
 	}
 
 	if tx.Commit().Error != nil {
 		tx.Rollback()
 		return installer.NewUpdateClusterInternalServerError().
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:522]: Don't we want this error in the log as well?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -290,10 +289,10 @@ func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inst
 			Errorf("Failed to get ISO: %s", imgName)
 		if resp.StatusCode == http.StatusNotFound {
 			return installer.NewDownloadClusterISONotFound().
-				WithPayload(generateError(http.StatusNotFound))
+				WithPayload(generateError(http.StatusNotFound, errors.New(string(b))))
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:292]: Dont we need b in the log as well?

Comment (filanov) [reply internal/bminventory/inventory.go:292]: it's already logged above 
```
		log.WithError(fmt.Errorf("%d - %s", resp.StatusCode, string(b))).
			Errorf("Failed to get ISO: %s", imgName)
```

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (filanov) [reply internal/bminventory/inventory.go:522]: you are right will add it.

**Pull request merged**